Kernel/Devices/HID/PS2KeyboardDevice.cpp

```diff
@@ -83,9 +83,11 @@ bool PS2KeyboardDevice::handle_irq(const RegisterState&)
 
 UNMAP_AFTER_INIT RefPtr<PS2KeyboardDevice> PS2KeyboardDevice::try_to_initialize(const I8042Controller& ps2_controller)
 {
-    auto device = adopt_ref(*new PS2KeyboardDevice(ps2_controller));
-    if (device->initialize())
-        return device;
+    auto keyboard_device_or_error = try_create_device<PS2KeyboardDevice>(ps2_controller);
+    // FIXME: Find a way to propagate errors
```

Comment:
item for the yak stack, one way to propogate errors from this function would be to have it return `KResultOr<NonnullRefPtr<PS2KeyboardDevice>>` and have init() on the device return KResult.
Then the body would become something like:

```c++
auto keyboard_device = TRY(try_create_device<PS2KeyboardDevice>(ps2_controller));
TRY(keyboard_device->initialize());
return keyboard_device.release_value();
```

This would apply to all the devices I think... most of these factories either return RefPtr<device> or NonnullRefPtr<Device>